apps/src/lib/util/GoogleAnalyticsReporter.js

```diff
@@ -0,0 +1,152 @@
+window.dataLayer = window.dataLayer || [];
+
+/**
+ * Global function used to track Google Universal events.
+ * It is a wrapper class that appends information to window.dataLayer
+ * This is the equivalent of ga() in Google Universal
+ */
+function gtag() {
+  window.dataLayer.push(arguments);
+}
+
+/**
+ * Class used to control the initialization of Google analytics
+ * for code.org properties.
+ * The class is responsible for
+ * - Adding the Google Tags needed by Tag manager
+ * - Send any dimensions information
+ * - Wrap Google Universal and Analytics functions for easier transition
+ * - Centralize common code
+ */
+class GoogleAnalyticsReporter {
+  enableGoogleAnalyticsUniversal = true;
+  enabledGoogleAnalytics4 = true;
+  googleUniversalGtag = 'UA-37745279-1';
+  googleAnalyticsGTag = 'G-L9HT5MZ3HD';
+
+  constructor() {
+    this.#initializeGoogleUniversal(
```

Comment:
Why are there `#`s?

Reply:
Is this the ES12 Private Class Fields?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

If so, it might not be compatible with our minimum Safari and Firefox versions:

https://code.org/educate/it

Reply:
yes, it is for private class. Should this not be included because of safari and firefox? :(

Reply:
We either need to not use it or make an argument to bump our minimum version :grin:

Reply:
we don't need to use it. I am just a fancy man with fancy needs. 
I can remove it and they will be public